qdrant/v1.2.x/configuration.md

```diff
@@ -3,20 +3,108 @@ title: Configuration
 weight: 160
 ---
 
-To change or correct Qdrant's behavior, default collection settings, and network interface parameters, you can use the configuration file.
+To change or correct Qdrant's behavior, default collection settings, and network interface parameters, you can use configuration files.
 
-Default configuration file is located in [config/config.yaml](https://github.com/qdrant/qdrant/blob/master/config/config.yaml).
+The default configuration file is located at [config/config.yaml](https://github.com/qdrant/qdrant/blob/master/config/config.yaml).
 
-In the production environment, you can override any value of this file by providing new values in `/qdrant/config/production.yaml` inside the docker.
+To change the default configuration, add a new configuration file and specify
+the path with `--config-path path/to/custom_config.yaml`. If running in
+production mode, you could also choose to overwrite `config/production.yaml`.
+See [ordering](#order-and-priority) for details on how configurations are
+loaded.
 
-Here is an example of how you can pass custom configuration inside the docker container:
+To use Qdrant in Docker and overwrite the production configuration use:
 
 ```bash
 docker run -p 6333:6333 \
     -v $(pwd)/path/to/custom_config.yaml:/qdrant/config/production.yaml \
     qdrant/qdrant
 ```
 
+Or use your own configuration file and specify it:
+
+```bash
+docker run -p 6333:6333 \
+    -v $(pwd)/path/to/custom_config.yaml:/qdrant/config/custom_config.yaml \
+    qdrant/qdrant \
+    ./qdrant --config-path config/custom_config.yaml
+```
+
+## Order and priority
+
+*Effective as of v1.2.1*
+
+Multiple configurations may be loaded on startup. All of them are merged into a
+single effective configuration that is used by Qdrant.
+
+Configurations are loaded in the following order, if present:
+
+1. Embedded base configuration ([source](https://github.com/qdrant/qdrant/blob/master/config/config.yaml))
+2. File `config/config.yaml`
+3. File `config/{RUN_MODE}.yaml` (such as `config/production.yaml`)
+4. File `config/local.yaml`
+5. Config provided with `--config-path PATH` (if set)
+6. [Environment variables](#environment-variables)
+
+This list is from least to most significant. Properties in later configurations
+will overwrite those loaded before it. For example, a property set with
+`--config-path` will overwrite those in other files.
+
+Most of these files are included by default in the Docker container. But it is
+likely that they are absent on your local machine if you run the `qdrant` binary
+manually.
+
+If file 2 or 3 are not found, a warning is shown on startup.
+If file 5 is provided but not found, an error is shown on startup.
+
+Other supported configuration file formats and extensions include: `.toml`, `.json`, `.ini`.
+
+To see the effective (merged) configuration you can query
+[telemetry](https://qdrant.tech/documentation/telemetry/):
+
+```bash
+curl 'http://localhost:6333/telemetry?details_level=6&anonymize=false'
```

Comment:
this gives the collection configuration but not the server configuration I believe

Reply:
Correct. I'm seeing the same result.

I think we do have to implement <https://github.com/qdrant/qdrant/issues/1973> then.